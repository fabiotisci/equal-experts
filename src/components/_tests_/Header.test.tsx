import * as React from 'react';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { Header } from '..';

configure({adapter: new Adapter()});

describe('<Header>', () => {
    let wrapper: ShallowWrapper;
    
    beforeEach( () => {
        wrapper = shallow(<Header />);
    });

    it('Should render the expected template', () => {
        expect(wrapper).toMatchSnapshot();
    });     
})