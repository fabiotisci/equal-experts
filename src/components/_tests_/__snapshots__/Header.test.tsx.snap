// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Should render the expected template 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="https://www.equalexperts.com/wp-content/themes/equalexperts/assets/logo.svg"
          style={
            Object {
              "maxWidth": "100px",
            }
          }
        />,
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          Calculator test
        </span>,
      ],
      "style": Object {
        "backgroundColor": "#1595d4",
        "color": "#FFF",
        "fontSize": "20px",
        "padding": "17px",
        "textAlign": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://www.equalexperts.com/wp-content/themes/equalexperts/assets/logo.svg",
          "style": Object {
            "maxWidth": "100px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Calculator test",
          "style": Object {
            "display": "block",
          },
        },
        "ref": null,
        "rendered": "Calculator test",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="https://www.equalexperts.com/wp-content/themes/equalexperts/assets/logo.svg"
            style={
              Object {
                "maxWidth": "100px",
              }
            }
          />,
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            Calculator test
          </span>,
        ],
        "style": Object {
          "backgroundColor": "#1595d4",
          "color": "#FFF",
          "fontSize": "20px",
          "padding": "17px",
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "https://www.equalexperts.com/wp-content/themes/equalexperts/assets/logo.svg",
            "style": Object {
              "maxWidth": "100px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Calculator test",
            "style": Object {
              "display": "block",
            },
          },
          "ref": null,
          "rendered": "Calculator test",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
